src/main/java/org/sunbird/workflow/service/UserBulkUploadService.java

```diff
@@ -629,4 +643,33 @@ private boolean validateCategory(String category) {
         return categoryValues != null && categoryValues.contains(category);
     }
 
+    private boolean validateGroupValue(String groupValue){
+        String groupValues = configuration.getGroupValues();
+        Set<String> groupValuesSet = Stream.of(groupValues.split(",")).collect(Collectors.toSet());
+        return groupValuesSet.contains(groupValue);
+    }
+
+    private boolean validateFieldValue(String fieldKey, String fieldValue){
```

Comment:
The approach in this method is performance consuming... it is converting list of string to comma separated string - everytime when the field validated, reads the string and creates a list by separating with comma -- which is unnecessary.
We have List of String value master_data -- we should store the same in the Redis -- refer hset / hget for the same.
Store list of string in Redis against the key -- just check how to verify the redis contains value for the key in it's list.

Reply:
This has been implemented as suggested.